src/local/room_position/approximate_offsets.rs

```diff
@@ -0,0 +1,147 @@
+//! Methods related to approximating positions between other positions.
+use crate::objects::HasPosition;
+
+use super::Position;
+
+impl Position {
+    /// Calculates an approximate midpoint between this point and the target.
+    ///
+    /// In case of a tie, rounds towards this point.
+    ///
+    /// If `distance_towards_target` is bigger than the distance to the target,
+    /// the target is returned.
+    pub fn towards<T>(self, target: &T, distance_towards_target: i32) -> Position
+    where
+        T: ?Sized + HasPosition,
+    {
+        let target = target.pos();
+
+        let (offset_x, offset_y) = target - self;
+        let total_distance = offset_x.abs().max(offset_y.abs());
+        if distance_towards_target > total_distance {
+            return target;
+        }
+
+        let new_offset_x = (offset_x * distance_towards_target) / total_distance;
+        let new_offset_y = (offset_y * distance_towards_target) / total_distance;
+
+        self + (new_offset_x, new_offset_y)
+    }
+
+    /// Calculates an approximate midpoint between this point and the target.
+    ///
+    /// In case of a tie, rounds towards the target.
+    ///
+    /// If `distance_from_target` is bigger than the distance to the target,
+    /// this position is returned.
+    pub fn between<T>(self, target: &T, distance_from_target: i32) -> Position
```

Comment:
Given that the other is `toward`, wouldn't this be more understandable as `from`?